modules/core/arrow-core-data/src/main/kotlin/arrow/core/ListK.kt

```diff
@@ -5,15 +5,130 @@ import arrow.higherkind
  *
- * A data type is considered a wrapper if
- * - It contains a single constructor with one parameter whose type is a type constructor with one type parameter
- * - It's a sub type of the type it wraps
+ * ank_macro_hierarchy(arrow.core.ListK)
+ *
+ * {:.beginner}
+ * beginner
+ *
+ * ListK wraps over the platform `List` type to make it a [type constructor](/docs/patterns/glossary/#type-constructors).
+ *
+ * It can be created from Kotlin List type with a convenient `k()` function.
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.k
+ *
+ * val value =
+ * //sampleStart
+ * listOf(1, 2, 3).k()
+ * //sampleEnd
+ * fun main() {
+ * println(value)
+ * }
+ * ```
+ *
+ * For most use cases you will never use `ListK` directly but `List` directly with the extension functions that Arrow projects over it.
+ *
+ * ListK implements operators from many useful typeclasses.
+ *
+ * The @extension type class processor expands all type class combinators that `ListK` provides automatically over `List`
+ *
+ * For instance, it has `combineK` from the [SemigroupK](/docs/arrow/typeclasses/semigroupk/) typeclass.
+ *
+ * It can be used to cheaply combine two lists:
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.extensions.list.semigroupK.combineK
+ *
+ * //sampleStart
+ * val hello = listOf('h', 'e', 'l', 'l', 'o')
+ * val commaSpace = listOf(',', ' ')
+ * val world = listOf('w', 'o', 'r', 'l', 'd')
+ *
+ * val combinedList = hello.combineK(commaSpace).combineK(world)
+ * //sampleEnd
+ * fun main() {
+ * println("combinedList = $combinedList")
+ * }
+ * ```
+ *
+ * The functions `traverse` and `sequence` come from [Traverse](/docs/arrow/typeclasses/traverse/).
+ *
+ * Traversing a list creates a new container [Kind<F, A>](/docs/patterns/glossary/#type-constructors) by combining the result of a function applied to each element:
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.None
+ * import arrow.core.Option
+ * import arrow.core.Some
+ * import arrow.core.extensions.list.traverse.traverse
+ * import arrow.core.extensions.option.applicative.applicative
+ *
+ * //sampleStart
+ * val numbers = listOf(Math.random(), Math.random(), Math.random())
+ * val traversedList = numbers.traverse(Option.applicative(), { if (it > 0.5) Some(it) else None })
+ * //sampleEnd
+ * fun main() {
+ * println("traversedList $traversedList")
+ * }
+ * ```
+ *
+ * and complements the convenient function `sequence()` that converts a list of `ListK<Kind<F, A>>` into a `Kind<F, ListK<A>>`:
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.Option
+ * import arrow.core.Some
+ * import arrow.core.extensions.list.traverse.sequence
+ * import arrow.core.extensions.option.applicative.applicative
+ *
+ * //sampleStart
+ * val requests = listOf(Some(Math.random()), Some(Math.random()), Some(Math.random()))
+ * val sequenceList = requests.sequence(Option.applicative())
+ * //sampleEnd
+ * fun main() {
+ * println("sequenceList = $sequenceList")
```

Comment:
```suggestion
 *   println("sequenceList = $sequenceList")
```

Reply:
Gotta make sure the kdocs are also properly formated as this examples show up in tooltips and other docs beside arrow